projects/components/src/description/description.component.ts

```diff
@@ -1,39 +1,78 @@
-import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  ViewChild,
+  AfterViewInit,
+  OnChanges,
+  ElementRef
+} from '@angular/core';
+import { Subject } from 'rxjs';
 
 @Component({
   selector: 'ht-description',
-  changeDetection: ChangeDetectionStrategy.OnPush,
   styleUrls: ['./description.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
-    <div class="description">
+    <div class="description" (htLayoutChange)="this.remeasure()" #eventDescriptionContainer>
       <div
         class="description-text"
-        [ngClass]="{ 'truncated-text': !this.isDescriptionTextToggled }"
+        [ngClass]="{ 'truncated-text': !isDescriptionTextToggled }"
         data-sensitive-pii
-        #eventDescription
+        #eventDescriptionText
       >
         {{ description }}
-        <span *ngIf="this.isDescriptionTextToggled" (click)="this.toggleDescriptionText()" class="description-button"
+        <span
+          *ngIf="(isDescriptionTruncated$ | async) && isDescriptionTextToggled"
+          (click)="toggleDescriptionText()"
+          class="description-button"
           >show less</span
         >
       </div>
       <div
-        class="description-button"
-        *ngIf="eventDescription.offsetWidth < eventDescription.scrollWidth && !this.isDescriptionTextToggled"
-        (click)="this.toggleDescriptionText()"
+        class="description-button description-button-more"
+        *ngIf="(isDescriptionTruncated$ | async) && !isDescriptionTextToggled"
+        (click)="toggleDescriptionText()"
       >
         show more
       </div>
     </div>
   `
 })
-export class DescriptionComponent {
+export class DescriptionComponent implements OnChanges, AfterViewInit {
+  public isInitialized: boolean = false;
   public isDescriptionTextToggled: boolean = false;
+  public readonly isDescriptionTruncated$: Subject<boolean> = new Subject();
```

Comment:
Is there an advantage of using an async boolean here (vs a simple boolean that is reassigned)?

Reply:
The original problem was *ngIf don't re-rendered after the change condition, so I decide to use this method. But now I'm use `ChangeDetectorRef` for this purpose